api/envoy/extensions/filters/http/router/v3/router.proto

```diff
@@ -75,7 +75,11 @@ message Router {
   }];
 
   // If not set, ingress Envoy will ignore
-  // :ref:`config_http_filters_router_x-envoy-expected-rq-timeout-ms` header, populated by egress
-  // Envoy, when deriving timeout for upstream cluster.
+  // :ref:`config_http_filters_router_x-envoy-expected-rq-timeout-ms` header,
+  // populated by egress Envoy, when deriving timeout for upstream cluster.
   bool respect_expected_rq_timeout = 6;
+
+  // Enable the addition of the `date` header to outbound requests.
+  // If `true`, the header will be assigned on each request (and retry).
```

Comment:
I think it's fine to add this, but per @PiotrSikora can you maybe add some text here about the potential misuse?

Also, I wonder if we should make this an enum with the default being off, a value being all requests and retries, with the option to later add a third enum that would add it on the initial request but not retries? That might also future proof it better.